satellite/audit/verifier.go

```diff
@@ -74,13 +78,22 @@ func NewVerifier(log *zap.Logger, metainfo *metainfo.Service, transport transpor
 	}
 }
 
-// Verify downloads shares then verifies the data correctness at the given stripe
-func (verifier *Verifier) Verify(ctx context.Context, stripe *Stripe, skip map[storj.NodeID]bool) (report *Report, err error) {
+// Verify downloads shares then verifies the data correctness at a random stripe.
+func (verifier *Verifier) Verify(ctx context.Context, path storj.Path, skip map[storj.NodeID]bool) (report *Report, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	pointer := stripe.Segment
+	pointer, err := verifier.checkIfSegmentDeleted(ctx, path)
```

Comment:
Replace this with loading the pointer from metainfo rather than calling `checkIfSegmentDeleted`. And keep the argument `oldPointer` for `checkIfSegmentDeleted`

Reply:
okay sounds good, thanks for the help!

Reply:
okay I did the above. this is looking difficult to test though. previously the test looked like this:
```
		err := ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
		require.NoError(t, err)

		satellite.Audit.Chore.Loop.TriggerWait()
		path, err := queue.Next()
		require.NoError(t, err)

		// replace the file
		err = ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
		require.NoError(t, err)

		report, err := audits.Verifier.Verify(ctx, path, nil)
		require.True(t, audit.ErrSegmentDeleted.Has(err))
		assert.Empty(t, report)
```
however, now in order for `audit.ErrSegmentDeleted.Has(err)` to be true, I think we would have to call Verify and the re-upload concurrently where the re-upload happens after Verify has started but before the `checkIfSegmentDeleted` call happens. do you know a way to test this that's not flaky?

Reply:
The best I can think of is adding a callback that can be set for testing, e.g. at the beggining of `checkIfSegmentDeleted` and in the callback then modify the metainfo database.

```
// checkIfSegmentDeleted checks if stripe's pointer has been deleted since stripe was selected.
func (verifier *Verifier) checkIfSegmentDeleted(ctx context.Context, segmentPath string, oldPointer *pb.Pointer) (newPointer *pb.Pointer, err error) {
	defer mon.Task()(&ctx)(&err)

	if verifier.OnTestingCheckSegmentDeletedHook != nil {
		verifier.OnTestingCheckSegmentDeletedHook
```

Reply:
added!